### PR TITLE
Druid Setup in docker-compose

### PR DESCRIPTION
**Pull request by Ideonvn**
TODO

- [X] Containerise Druid in docker-compose
- [X] Load custom Zenysis extensions
- [X] Updated environments file
- [x] Indexing
- [x] Caching
- [x] Druid coordinator
- [x] Assorted

### CODE REVIEW

File: druid_setup/environment [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+druid_indexer_fork_property_druid_processing_buffer_sizeBytes=256MiB
+
+druid_storage_type=local
+druid_storage_storageDirectory=/opt/shared/segments
+druid_indexer_logs_type=file
+druid_indexer_logs_directory=/opt/shared/indexing-logs
+
+druid_processing_numThreads=2
+druid_processing_numMergeBuffers=2
+
```
Comment (AbbyBeeler) [inline druid_setup/environment:53]: Other assorted settings:

`druid_segmentCache_locations` is where historical puts the local file cache for active datasources and is a required property. It should be put in a volume (the actual path doesn't matter), but only historical needs access to it

```
druid_javascript_enabled=true
druid_coordinator_asOverlord_enabled=true
druid_coordinator_asOverlord_overlordService=druid/overlord
druid_indexer_runner_type=remote
druid_indexer_storage_type=metadata
druid_segmentCache_locations=[{"path": "/data/output/druid", "maxSize": 300000000000, "freeSpacePercent": 5}]
```

File: druid_setup/environment [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#
```
Comment (AbbyBeeler) [inline druid_setup/environment:1]: Druid recommends splitting out the environment files, do we want to do that? https://druid.apache.org/docs/latest/tutorials/docker.html#environment-file

**Review by AbbyBeeler — COMMENTED** [review]
I left my initial thoughts. Druid recommends structuring the environment file differently for production setups and says that production setups will need different defaults (https://druid.apache.org/docs/latest/tutorials/docker.html#configuration). Since I think Brazil and most places will basically run druid as provided, I think we should do that ourselves.

Also, I think we may be able to exclude the hadoop dependency, I'm going to play around with that on Monday.

File: druid_setup/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+
+services:
+  extension_loader:
+    build:
+      context: extensions
+    volumes:
+      - zen_extensions:/druid/extensions
+    environment:
+      - ZEN_DRUID_VERSION=0.22.1
```
[the remark this answers is not in the document]
Comment (Ideonvn) [reply druid_setup/docker-compose.yml]: Indeed we should, done!

[the remark this answers is not in the document]
Comment (Ideonvn) [reply druid_setup/docker-compose.yml]: Done

File: druid_setup/extensions/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,10 @@
+FROM alpine:latest
+
+RUN apk add bash
+
+WORKDIR /druid/extensions
+
+ENV ZEN_DRUID_VERSION="0.22.1"
```
[the remark this answers is not in the document]
Comment (Ideonvn) [reply druid_setup/extensions/Dockerfile]: This is more a default making the env optional (was more relevant for testing, I've made it required rather)

**Pull request merged**